protected/views/dataset/view.php

```diff
@@ -35,7 +35,12 @@
                                 <p><?= $mainSection->getReleaseDetails()['authors'] ?> (<?=$mainSection->getReleaseDetails()['release_year']?>): <?= $mainSection->getReleaseDetails()['dataset_title'].' '.($mainSection->getReleaseDetails()['publisher'] ?? '<span class="label label-danger">NO PUBLISHER SET</span>').'. '; ?><a href="https://doi.org/10.5524/<?php echo $model->identifier;?>">https://doi.org/10.5524/<?php echo $model->identifier;?></a></p>
                                 <p><a class="doi-badge" href="#"><span class="badge">DOI</span><span class="badge">10.5524/<?php echo $model->identifier;?></span></a><button onclick="showCitation()" class="drop-citation-btn" >Cite Dataset<span class="caret"></span></button></p>
                                 <div id="citationDropdown" class="citation-content">
-                                    <a id="citeText" href='https://data.datacite.org/text/x-bibliography/10.5524/<?php echo $model->identifier;?>' target="_blank">Text</a>
+                                    <?php
+                                    $identifier = $model->identifier;
+                                    $text = file_get_contents('https://data.datacite.org/text/x-bibliography/10.5524/'.$identifier);
+                                    $clean_text = strip_tags(preg_replace("/&#?[a-z0-9]+;/i","", $text));
+                                    ?>
+                                    <a id="citeText" onclick="showText()">Text</a>
```

Comment:
Hi @pli888 

I removed the `html entities` using `preg_replace` and `strip_tags`, then show the `clean_text` in new window using  onclick `showText()` function.
The function works as expected as below:
![image](https://user-images.githubusercontent.com/64770635/104274093-0dbd1a80-54db-11eb-80fd-f8c62ecc14a9.png)

Do you think it is a proper way to implement a method like this?

Reply:
Hi @kencho51 ,
I'm not sure where this little screenshot comes from or if I'm missing the point of this thread entirely :-), but the DOI 10.5524/100005 is a GigaDB DOI so should be shown as publisher "GigaScience Database" not just "GigaScience". This could perhaps be an issue with the data in the dev database?
As for HTML tags in it, ideally the species name would be displayed in italics which is why the html code would have been included in the title originally. but its not the end of the world if its not italicised.

Reply:
Hi @only1chunts ,

In the `citation dropdown` box, one of the option `Text` would return `x-bibliography` from `crosscite` in a new tab when you click on it, for example:
```
Li, J., Zhang, G., Lambert, D., &amp; Wang, J. (2011). <i>Genomic data from the Emperor penguin (Aptenodytes forsteri)</i> [Data set]. GigaScience. https://doi.org/10.5524/100005
```
Then above return contains `&amp` and `<i></i>` which should be removed for text citation?

This is the link to generate `crosscite` citation in x-bibliography format:
```
https://data.crosscite.org/text/x-bibliography/10.5524/100005
```

Reply:
Ah I see the issue, its a very old dataset which was published before we set the publisher name to "GigaScience Database" thats why its showing as just "GigaScience", thats one issue solved!
for the html codes, that is stuff that CrossCite are adding into our metadata, because the call "https://data.crosscite.org/text/x-bibliography/10.5524/100005" is specifically asking for a formatted text citation, i.e. your call is telling it to put the formatting in there.
Maybe it would be better to label the item as "formatted text" rather than just "text" and leave it as its given from CrossCite

Reply:
Hi @only1chunts ,

So you want something like this?
![image](https://user-images.githubusercontent.com/64770635/104437166-6f5bb280-55c9-11eb-9fb6-bce639973980.png)

Reply:
that looks perfect, thanks.